streams/src/main/java/org/apache/kafka/streams/processor/StateStoreSupplier.java

```diff
@@ -34,14 +36,15 @@
     /**
      * Return a new {@link StateStore} instance.
      *
-     * @return  a new {@link StateStore} instance
+     * @return a new {@link StateStore} instance of type T
      */
-    StateStore get();
+    T get();
```

Comment:
@guozhangwang  it is so we at least try tor enforce users to pass the correct type of StateStoreSupplier to the methods in `KGroupedStream`. i.e., it is not valid to pass a `StateStoreSupplier` that creates a `KeyValueStore` to an operation expecting a `WindowStore`, and vice-versa.

example below:
```
<T> KTable<K, T> aggregate(Initializer<T> initializer,
                               Aggregator<K, V, T> aggregator,
                               final StateStoreSupplier<KeyValueStore> storeSupplier);
```
and
```
<W extends Window, T> KTable<Windowed<K>, T> aggregate(Initializer<T> initializer,
                                                           Aggregator<K, V, T> aggregator,
                                                           Windows<W> windows,
                                                           final StateStoreSupplier<WindowStore> storeSupplier);
```